sources/models/src/modeled_types/ecs.rs

```diff
@@ -228,3 +228,57 @@ mod test_ecs_agent_log_level {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// ECSAgentImagePullBehavior represents a string that contains a valid ECS Image Pull Behavior for the ECS agent.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ECSAgentImagePullBehavior {
+    inner: String,
+    pub inner_code: String,
+}
```

Comment:
My concern with this representation is that it implies we'll end up serializing both the string and the code to the datastore, but the `string_impls_for!` macro only handles a single `inner` field in its serialize impl (and we only want it to).

Instead I'd suggest adding a `TryFrom<&str>` impl for `ECSImagePullBehavior`, which should give you a `TryInto` impl that you can use to convert the string into an enum value, which you can then cast to `u8`.

Reply:
It's not obvious or very common, but we do make use of the modeled types in other programs. Here's the gist of [what I had in mind](https://gist.github.com/bcressey/0d52bff9ec7ec23d68f9556cf47b5a58).

Essentially:
* add `TryFrom<&str>` impl to `ECSImagePullBehavior` (and reuse it in `ECSAgentImagePullBehavior`)
* make `ECSImagePullBehavior` pub, since we want its `TryFrom` impl outside the crate
* add a dependency on the `models` crate to `ecs-settings-applier`
* use the `TryFrom` impls to turn the string into the enum, and then cast as a u8

(The `unwrap()` is a bit of a smell but we know we haven't changed the value and that it was verified during deserialization, and it's awkward to go from an Option to a Result to an Option in a function chain.)

Reply:
Thanks for the clear explanation and for the code.
I just made a push that include it and your other suggestion too.
If you want to do changes or write it down in a better way feel free to do it.